tools/automatic_query_fixer/src/main/java/com/google/cloud/bigquery/utils/queryfixer/errors/SqlErrorFactory.java

```diff
@@ -0,0 +1,136 @@
+package com.google.cloud.bigquery.utils.queryfixer.errors;
+
+import com.google.cloud.bigquery.BigQueryException;
+import com.google.cloud.bigquery.utils.queryfixer.entity.Position;
+import com.google.cloud.bigquery.utils.queryfixer.util.PatternMatcher;
+
+import java.util.List;
+
+/**
+ * A factory to create {@link BigQuerySqlError} from {@link BigQueryException}. Currently, it
+ * supports {@link TableNotFoundError}, {@link UnrecognizedColumnError}, and {@link
+ * FunctionNotFoundError}.
+ */
+public class SqlErrorFactory {
+
+  public static final String TableNotFoundRegex = "^Not found: Table (.*?) was not found";
+
+  public static final String UnrecognizedNameRegex =
+      "^Unrecognized name: (.*?)(; Did you mean (.*?)\\?)? at (.*?)$";
+
+  public static final String FunctionNotFoundRegex =
+      "^Function not found: (.*?)(; Did you mean (.*?)\\?)? at (.*?)$";
+
+  /**
+   * The method to convert {@link BigQueryException} to {@link BigQuerySqlError}. If the input
+   * exception can not be resolved, a null pointer will be returned instead.
+   *
+   * @param exception the BigQueryException to resolve.
+   * @return BigQuerySqlError object or null pointer.
+   */
+  public BigQuerySqlError getError(BigQueryException exception) {
+    BigQuerySqlError error;
+
+    if ((error = tryTableNotFoundError(exception)) != null) {
```

Comment:
I'm not sure if this is the recommended practice to save one line, but up to you :)

Reply:
Let's keep in this style and change it later if the style is discouraged.